migrating.md

```diff
@@ -96,6 +100,39 @@ It's often prohibitively time-consuming to test every JavaScript interaction on
+- JS error like `something is not defined` where `something` is an HQ module: this can happen when script tags are ordered so that a module appears before one of its dependencies. This can happen to migrated modules because one of the effects of the migration is to typically import all of a module's dependencies at the time the module is defined, which in a non-RequireJS context means all of the dependencies' script tags must appear before the script tags that depend on them. Previously, dependencies were not imported until `hqImport` was called, which could be later on, possibly in an event handler or some other code that would never execute until the entire page was loaded. To fix, try reordering the script tags. If you find there's a circular dependency, use `hqRequire` as described above.
+
+#### Troubleshooting the RequireJS build process
+
+Tactics that can help track down problems with the RequireJS build process, which usually manifest as errors that happen on staging but not locally:
+
+- To turn off minification, add an `optimize: none` line to [requirejs.yaml](https://github.com/dimagi/commcare-hq/tree/master/corehq/apps/hqwebapp/static/hqwebapp/yaml/requirejs.yaml).
+- To stop using the CDN, comment out [resource_versions.js in hqwebapp/base.html](https://github.com/dimagi/commcare-hq/pull/18116/files#diff-1ecb20ffccb745a5c0fc279837215a25R433). Note that this will still fetch a few files, such as `hqModules.js` and `requirejs_config.js`, from the CDN. To turn off the CDN entirely, comment out all of the code that manipulates `resource_versions` in [build_requirejs](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/hqwebapp/management/commands/build_requirejs.py).
+- To mimic the entire build process locally:
+   - Collect static files: `manage.py collectstatic --noinput`  This is necessary if you've made any changes to `requirejs.yaml` or `requirejs_config.js`, since the build script pulls these files from `staticfiles`, not `corehq`.
+   - Compile translation files: `manage.py compilejsi18n`
+   - Run build script: `manage.py build_requirejs`
+   - Copy build-generated files back into `corehq` since you're in a development environment:
+      - Updated config, including bundle setup: `cp staticfiles/hqwebapp/js/requirejs_config.js corehq/apps/hqwebapp/static/hqwebapp/js/requirejs_config.js`
+      - CDN paths: `cp staticfiles/hqwebapp/js/resource_versions.js corehq/apps/hqwebapp/static/hqwebapp/js/resource_versions.js`
+      - Bundles
+         - `cp staticfiles/hqwebapp/js/common.js corehq/apps/hqwebapp/static/hqwebapp/js/common.js`
+         - `cp staticfiles/hqwebapp/js/bundle.js   corehq/apps/hqwebapp/static/hqwebapp/js/bundle.js`
+         - ...and the same for any other bundles used on the page you're testing. The easiest way to check this is to open the page on staging, open the browser console, and check the resources tab to see what javascript files are requests.
```

Comment:
nit/typo: requests -> requested

at least I read all the way to the last word 😛

Reply:
I do appreciate that ;)